test/features/mailbox/repository/mailbox_respository_test.dart

```diff
@@ -229,4 +236,47 @@ void main() {
       );
     });
   });
+
+  group('[handleMailboxRightRequest] method test', () {
+    const postingRight = 'p';
+
+    setUp(() {
+      mailboxDataSource = MockMailboxDataSource();
+      mailboxCacheDataSourceImpl = MockMailboxCacheDataSourceImpl();
+      stateDataSource = MockStateDataSource();
+      mailboxRepository = MailboxRepositoryImpl(
+          {
+            DataSourceType.network: mailboxDataSource,
+            DataSourceType.local: mailboxCacheDataSourceImpl
+          },
+          stateDataSource);
+    });
+
+    test(
+      'handleMailboxRightRequest SHOULD add postingRight when action is allow and currentRights is empty', () async {
+        Map<String, List<String>?> currentRights = {};
+        MailboxRightRequest request = MailboxRightRequest(mailboxIdFixture, currentRights, MailboxSubaddressingAction.allow);
+
+        when(stateDataSource.getState(accountIdFixture, userNameFixture, StateType.mailbox))
+            .thenAnswer((_) async => StateFixtures.currentMailboxState);
+
+        when(mailboxDataSource.handleMailboxRightRequest(sessionFixture, accountIdFixture, request))
+          .thenAnswer((_) async => true);
+
+        when(mailboxDataSource.getChanges(sessionFixture, accountIdFixture, StateFixtures.currentMailboxState))
+          .thenAnswer((_) async => MailboxChangeResponse(
+              // dont really know what to put in there
+              updated: [mailboxFixture],
+              updatedProperties: Properties({'rights'})
+        ));
+
+        final response = await mailboxDataSource.getChanges(
+          sessionFixture,
+          accountIdFixture,
+          StateFixtures.currentMailboxState,
+        );
+
+        expect(response.props[6], Properties({'rights'}));
```

Comment:
here i want to write a test for `updateRightsForSubaddressing`, which will be turned private, so the test is done via the caller method `handleMailboxRightRequest`.

i tried to do something based on what i could find, but i dont really understand how these kind of tests work (i am basically expecting on line 279 what i am inputing on line 269, so of course it works but it does not test anything...)

Reply:
dat dang suggested to add this in order to be able to test `updateRightsForSubaddressing` directly, it works fine if you're ok with it:

```
@visibleForTesting
List<String> updateRightsForSubaddressing(MailboxSubaddressingAction action, List<String>? currentRights)
    => _updateRightsForSubaddressing(action, currentRights);
```